src/libraries/System.Linq/src/System/Linq/Skip.cs

```diff
@@ -30,7 +30,7 @@ public static IEnumerable<TSource> Skip<TSource>(this IEnumerable<TSource> sourc
 
                 count = 0;
             }
-            else if (!IsSizeOptimized && source is Iterator<TSource> iterator)
+            else if (source is Iterator<TSource> iterator)
```

Comment:
Do you need a similar change in Enumerable.Take? As it's currently implemented, I don't think it'll end up using the Take method on your new iterator type.